garrysmod/gamemodes/terrortown/gamemode/vgui/sb_main.lua

```diff
@@ -132,35 +197,129 @@ function PANEL:Init()
-      self:AddColumn( GetTranslation("sb_karma") )
+      self:AddColumn( GetTranslation("sb_karma"), nil, nil,     "karma" )
    end
 
-   -- Let hooks add their column headers (via AddColumn())
+   self.sort_headers = {}
+   -- Reuse some translations
+   self:AddFakeColumn( GetTranslation("sb_sortby"), nil, nil,       nil ) -- "Sort by:"
+   self:AddFakeColumn( GetTranslation("equip_spec_name"), nil, nil, "name" )
+   self:AddFakeColumn( GetTranslation("col_role"), nil, nil,        "role" )
+  
+   -- Let hooks add their column headers (via AddColumn() or AddFakeColumn())
    hook.Call( "TTTScoreboardColumns", nil, self )
-
+  
    self:UpdateScoreboard()
    self:StartUpdateTimer()
 end
 
--- For headings only the label parameter is relevant, func is included for
+local function sort_header_handler(self_, lbl)
+   return function()
+      surface.PlaySound("ui/buttonclick.wav")
+
+      local sorting = GetConVar("ttt_scoreboard_sorting")
+      local ascending = GetConVar("ttt_scoreboard_ascending")
+
+      if lbl.HeadingIdentifier == sorting:GetString() then
+         ascending:SetBool(not ascending:GetBool())
+      else
+         sorting:SetString( lbl.HeadingIdentifier )
+         ascending:SetBool(true)
+      end
+
+      for _, scoregroup in pairs(self_.ply_groups) do
+         scoregroup:UpdateSortCache()
+         scoregroup:InvalidateLayout()
+      end
+
+      self_:ApplySchemeSettings()
+   end
+end
+
+-- For headings only the label parameter is relevant, second param is included for
 -- parity with sb_row
-function PANEL:AddColumn( label, func, width )
-   local lbl = vgui.Create( "DLabel", self )
+local function column_label_work(self_, table_to_add, label, _, width, sort_identifier, sort_func )
+   local lbl = vgui.Create( "DLabel", self_ )
    lbl:SetText( label )
+   local can_sort = false
    lbl.IsHeading = true
    lbl.Width = width or 50 -- Retain compatibility with existing code
 
-   table.insert( self.cols, lbl )
+   if sort_identifier != nil then
+      can_sort = true
+      -- If we have an identifier and an existing sort function then it was a built-in
+      -- Otherwise...
+      if self_.sort_table[sort_identifier] == nil then
+         if sort_func == nil then
+            ErrorNoHalt( "Sort ID provided without a sorting function, Label = ", label, " ; ID = ", sort_identifier )
+            can_sort = false
+         else
+            self_.sort_table[sort_identifier] = sort_func
+         end
+      end
+   end
+
+   if can_sort then
+      lbl:SetMouseInputEnabled(true)
+      lbl:SetCursor("hand")
+      lbl.HeadingIdentifier = sort_identifier
+      lbl.DoClick = sort_header_handler(self_, lbl)
+   end
+
+   table.insert( table_to_add, lbl )
    return lbl
 end
 
+-- Apparently the following doesnt work
+-- PANEL.AddColumn = column_label_work(self.cols)
+function PANEL:AddColumn( label, func, width, sort_id, sort_func ) 
+   column_label_work( self, self.cols, label, func, width, sort_id, sort_func )

```

Comment:
Also, remove the comment. It doesn't work since you're trying to set PANEL.AddColumn to a return value, not a function.